doc/decisions/DR020-Coding-style.md

```diff
@@ -0,0 +1,68 @@
+# DR020 Coding style
+
+- **Status:** Decided
+- **Driver:** @foundrytom
+- **Approver:** @felltech @carmenpinto @fnRaihanKibria
+- **Outcome:** OpenAssetIO will use the Google C++ Style Guide.
+
+## Background
+
+OpenAssetIO is a collaboratively maintained codebase. A consistent
+coding style can help simplify maintenance. As such, contributors need
+clear guidelines.
+
+## Options considered
+
+### Option 1 - Define a custom style guide
+
+Maintainers define a bespoke style guide that embodies their
+locally agreed preferences.
+
+#### Pros
+
+- Allows local consensus around what is preferable in the code base.
+
+#### Cons
+
+- Consensus is nigh on impossible when it comes to subjective choices
+  such as code style, as the number of contributors grows beyond one.
+  Often resulting in unbalanced compromise.
+- Disagreements over the style may well be directly attributable to a
+  specific individual within the community, that then becomes the target
+  of criticism.
+- Local control over style minutiae adds the temptation to futz with it
+  over time, ultimately reducing overall consistency.
+- May increase the complexity of linter configuration when attempting
+  to encode the agreed standard.
+
+### Option 2 - Use a common well-established standard
+
+Pick a well known existing standard and just use that.
+
+#### Pros
+
+- Significantly reduces the complexity of the decision space as it is
+  now simply picking one of several presets instead of every aspect of a
+  style guide.
+- The details of the style are not owned by the project, deflecting
+  disagreements to an abstract third party, and removing the temptation
+  to futz.
+- Likely that most contributors will have some (individual)
+  disgruntlement with the style, sharing the feeling of compromise.
+- If a popular standard is chosen, it may be supported out-the-box by
+  linters.
+
+#### Cons
```

Comment:
[consider]

* A style guide (and tooling enforcing it) does not necessarily cover everything. This can lead to a false sense of security, in turn leading to growing inconsistencies that are not addressed.
* Risk that some sensible exceptions to the style guide, which could improve team effectiveness, may not be considered.
* Style guides are invariably language-specific. Languages in a multi-language project risk being either conformed to a non-idiomatic style, or made inconsistent with other languages.

Reply:
Are the first and last points true of both options?

Reply:
Added the second two [https://github.com/OpenAssetIO/OpenAssetIO/pull/1096/commits/b4e33b673288fc113cbf740060a6cb9e739e1e2b](here).

Reply:
For the first point I reasoned that a hand-rolled style guide makes you constantly think about unhandled cases and enforce that in the config over time.  Whereas a pre-rolled style guide you might assume covers everything, so as long as you don't get linter/formatter errors you assume it matches the style, when actually its inconsistent.

I had in my mind grouping/ordering of `#include` statements, which is only partially enforced via cpplint and clang-format.